andes/variables/dae.py

```diff
@@ -473,7 +473,7 @@ def set_t(self, t):
         Helper function for setting time in-place.
         """
 
-        self.t.itemset(t)
```

Comment:
This function is meant to change `self.t` in place. The new implementation creates a new `np.array` for `self.t`. I can't recall other usages of `self.t` that assume a persistent memory. If that exists, the new implementation will fail: even though `t` is being updated here, the previous memory is being referred elsewhere.

Can you check: 1) if this function is called elsewhere? 2) any other implementation that changes `t` in place? I know numpy scalars are immutable; that's why `itemset` was used.

Reply:
`itemset` is only used in the `reset()` method:
https://github.com/CURENT/andes/blob/5ab784b75347a98d07c004e32d7c4b39d407ea2e/andes/variables/dae.py#L458-L476

I changed to in-place assignment, and the test shows below:
```python
print(f"Memory address before: {ss.dae.t.ctypes.data}")
ss.dae.reset()
print(f"Memory address after: {ss.dae.t.ctypes.data}")

Memory address before: 4393601808
Memory address after: 4393601808
```

Reply:
Thanks. Looks good! I didn't know triple dots do the trick!

Reply:
I learnt this from GitHub Copilot just right now, a positive AGI datapoint!